### PR TITLE
Hyper sync rustls 0.6.1

### PR DESCRIPTION
**Pull request by SimranGujral**
This change removes Hyper-OpenSSL and integrates HyperSyncRustls v 0.6.1 with servo.
It also removes ciphers depending on DHE and SHA1

---

- [X] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17938)

### CODE REVIEW

File: components/net/connector.rs [hunk 1]
```diff
@@ -50,35 +50,20 @@ impl NetworkConnector for HttpsConnector {
...
+pub fn create_ssl_client(ca_file: &PathBuf) -> TlsClient {
+    let mut ca = {
+        let f = fs::File::open(ca_file).unwrap();
+        let rd = io::BufReader::new(f);
+        rd
+    };
+    let mut tls = rustls::ClientConfig::new();
+    tls.root_store.add_pem_file(&mut ca).unwrap();
+    let tls = Arc::new(tls);
+    let client = TlsClient { cfg: tls };
```
Comment (avadacatavra) [inline components/net/connector.rs]: `TlsClient { cfg: Arc::new(tls) }`
Rust automatically returns the last expression of a function: https://rustbyexample.com/fn.html

**Review by avadacatavra — CHANGES_REQUESTED** [review]

**Comment by avadacatavra** [comment]
This also needs to be squashed into one commit -- we can go over that tomorrow

http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Comment by SimranGujral** [comment]
All changes have been made and are ready for review.

File: components/net/connector.rs [hunk 2]
```diff
@@ -1,32 +1,31 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
```
Comment (avadacatavra) [inline components/net/connector.rs]: nit: put this line back in

File: components/net/connector.rs [hunk 3]
```diff
@@ -50,35 +49,17 @@ impl NetworkConnector for HttpsConnector {
...
-    {
-        let context = ssl_connector_builder.builder_mut();
-        context.set_ca_file(ca_file).expect("could not set CA file");
-        context.set_cipher_list(DEFAULT_CIPHERS).expect("could not set ciphers");
-        context.set_options(SSL_OP_NO_SSLV2 | SSL_OP_NO_SSLV3 | SSL_OP_NO_COMPRESSION);
-    }
-    let ssl_connector = ssl_connector_builder.build();
-    OpensslClient::from(ssl_connector)
+pub fn create_ssl_client(ca_file: &PathBuf) -> TlsClient {
+    let mut ca = {
```
Comment (avadacatavra) [inline components/net/connector.rs]: `let mut ca = {
let f = fs::File::open(ca_file).expect("cannot open CA file");
io::BufReader::new(f)
}`

File: components/net/lib.rs [hunk 4]
```diff
@@ -5,15 +5,14 @@
 #![deny(unsafe_code)]
 #![feature(box_syntax)]
 #![feature(iterator_step_by)]
-
```
Comment (avadacatavra) [inline components/net/lib.rs]: nit: put this line back in

**Review by avadacatavra — CHANGES_REQUESTED** [review]
The squash is good--I would clean up the commit comments to be more succinct. Currently they restate the same thing multiple times

**Comment by SimranGujral** [comment]
Changes have been made. Ready for review.

**Comment by jdm** [comment]
`test fetch::test_fetch_with_hsts has been running for over 60 seconds`
That's worrying! I bet that running any HTTPS WPT test will end up timing out.
@bors-servo: try

**Comment by jdm** [comment]
```
	thread 'fetch::test_fetch_with_hsts' panicked at 'called `Option::unwrap()` on a `None` value', /checkout/src/libcore/option.rs:335:20
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
             at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at /checkout/src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at /checkout/src/libstd/sys_common/backtrace.rs:60
             at /checkout/src/libstd/panicking.rs:380
   3: std::panicking::default_hook
             at /checkout/src/libstd/panicking.rs:390
   4: std::panicking::rust_panic_with_hook
             at /checkout/src/libstd/panicking.rs:611
   5: std::panicking::begin_panic_new
             at /checkout/src/libstd/panicking.rs:553
   6: std::panicking::begin_panic_fmt
             at /checkout/src/libstd/panicking.rs:521
   7: rust_begin_unwind
             at /checkout/src/libstd/panicking.rs:497
   8: core::panicking::panic_fmt
             at /checkout/src/libcore/panicking.rs:71
   9: core::panicking::panic
             at /checkout/src/libcore/panicking.rs:51
  10: <core::option::Option<T>>::unwrap
             at /checkout/src/libcore/macros.rs:22
  11: net_tests::fetch::test_fetch_with_hsts
             at ./fetch.rs:557
  12: <F as test::FnBox<T>>::call_box
             at /checkout/src/libtest/lib.rs:1477
             at /checkout/src/libcore/ops/function.rs:143
             at /checkout/src/libtest/lib.rs:138
  13: __rust_maybe_catch_panic
             at /checkout/src/libpanic_unwind/lib.rs:98
```
WPT tests worked, so that's good.  The android build appears to break while building `ring`.

**Comment by avadacatavra** [comment]
@jdm my suspicion is that we'll need to specify android to ring in the build command based on https://github.com/briansmith/ring/blob/master/build.rs#L613

**Comment by jdm** [comment]
The target is provided automatically since we're invoking Cargo with --target, and the C commands have android-specific include directories appended to them:
```
running "arm-linux-androideabi-gcc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "--sysroot" "/home/servo/android/ndk/current/platforms/android-18/arch-arm" "-I/home/servo/android/ndk/current/sources/android/support/include" "-g" "-march=armv7-a" "-march=armv7-a" "-mfpu=vfpv3-d16" "-mfloat-abi=softfp" "-I" "include" "-std=c1x" "-Wbad-function-cast" "-Wmissing-prototypes" "-Wnested-externs" "-Wstrict-prototypes" "-fdata-sections" "-ffunction-sections" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-declarations" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-Wno-cast-align" "-fstack-protector" "-g3" "-D_XOPEN_SOURCE=700" "-D__ANDROID_API__=18" "-c" "-o/home/servo/buildbot/slave/android/build/target/armv7-linux-androideabi/debug/build/ring-a621dfbd7d7ae5c7/out/ecp_nistz256.o" "crypto/ec/ecp_nistz256.c"
```

**Comment by jdm** [comment]
https://github.com/briansmith/ring/blob/5f8d332ed0f38415ee003712d1bb93647ed34afe/build.rs#L560 is the line that is adding the include directory that contains the standard library headers that clang does not like:
```
In file included from include/GFp/base.h:60:0,
                 from include/GFp/bn.h:126,
                 from crypto/bn/montgomery_inv.c:15:
/home/servo/android/ndk/current/sources/android/support/include/stdint.h:31:2: error: #include_next is a GCC extension
 #include_next <stdint.h>
  ^
thread '<unnamed>' panicked at 'execution failed', /home/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.11.0/build.rs:658:8
```

**Comment by SimranGujral** [comment]
The android issue is still in progress, Other issues have been resolved. Ready for review.
r? @avadacatavra

**Review by avadacatavra — APPROVED** [review]

**Comment by jdm** [comment]
`test fetch::test_fetch_with_hsts has been running for over 60 seconds`. TravisCI still does not like running that test now.

**Comment by Manishearth** [comment]
Apparently if we pass `--disable-werror` as a cflag when compiling this should go away.

But I'm not sure, since the warning doesn't actually mention Werror.

**Comment by jdm** [comment]
@Manishearth What should go away?

**Comment by Manishearth** [comment]
Oh. The GCC extension warning.

**Comment by avadacatavra** [comment]
I'm going to close this until I have more time to bring it up to date :)

**Pull request closed without merging**